met-web/src/components/engagement/form/EngagementWidgets/WidgetCardSwitch.tsx

```diff
@@ -1,26 +1,35 @@
+import { DragItem } from 'components/common/Dragndrop';
 import { Widget, WidgetType } from 'models/widget';
-import React from 'react';
+import React, { useContext } from 'react';
 import { Switch, Case } from 'react-if';
 import MetWidget from './MetWidget';
+import { WidgetDrawerContext } from './WidgetDrawerContext';
 
 interface WidgetCardSwitchProps {
     widget: Widget;
+    index: number;
+    moveWidget: (dragIndex: number, hoverIndex: number) => void;
+    removeWidget: (widgetId: number) => void;
 }
-export const WidgetCardSwitch = ({ widget }: WidgetCardSwitchProps) => {
+export const WidgetCardSwitch = ({ widget, index, moveWidget, removeWidget }: WidgetCardSwitchProps) => {
+    const { handleWidgetDrawerOpen } = useContext(WidgetDrawerContext);
+
     return (
         <>
             <Switch>
                 <Case condition={widget.widget_type_id === WidgetType.WhoIsListening}>
-                    <MetWidget
-                        testId={`who-is-listening-${widget.widget_type_id}`}
-                        title="Who is Listening"
-                        onDelete={() => {
-                            /**/
-                        }}
-                        onEdit={() => {
-                            /**/
-                        }}
-                    />
+                    <DragItem name="Who is Listening" moveItem={moveWidget} index={index}>
+                        <MetWidget
+                            testId={`who-is-listening-${widget.widget_type_id}`}
+                            title="Who is Listening"
+                            onDelete={() => {
+                                removeWidget(widget.id);
+                            }}
+                            onEdit={() => {
+                                handleWidgetDrawerOpen(true);
```

Comment:
consider opening the drawer and setting the tab to the who is listening form. Check if that works

Reply:
great idea it worked